src/utils/formatPath.ts

```diff
@@ -1,51 +1,40 @@
-export const getBasePathGenerator = () => {
-    let basePath: string | undefined;
-    const DEFAULT = '::baseUriPath::';
-
-    return () => {
-        if (process.env.NODE_ENV === 'development') {
-            return '';
-        }
-
-        if (basePath !== undefined) {
-            return basePath;
-        }
-        const baseUriPath = document.querySelector<HTMLMetaElement>(
-            'meta[name="baseUriPath"]'
-        );
-
-        if (baseUriPath?.content) {
-            basePath = baseUriPath?.content;
-
-            if (basePath === DEFAULT) {
-                basePath = '';
-                return '';
-            }
-
-            return basePath;
-        }
-        basePath = '';
-        return basePath;
-    };
+export const formatApiPath = (path: string, base = basePath): string => {
+    return joinPaths(base, path);
 };
 
-export const getBasePath = getBasePathGenerator();
+export const formatAssetPath = (path: string, base = basePath): string => {
```

Comment:
I'm a bit wary of changing the basePath logic. Why was this needed?

Reply:
Good point. Needed to run localhost with a basePath to test the instance endpoint, and took the time to refactor a bit.

https://github.com/Unleash/unleash-frontend/blob/trial-status/vite.config.ts#L8